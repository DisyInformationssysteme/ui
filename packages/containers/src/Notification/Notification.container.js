import PropTypes from 'prop-types';
import React from 'react';
import { List, Map } from 'immutable';
import Notification from '@talend/react-components/lib/Notification';
import { cmfConnect } from '@talend/react-cmf';

export const DEFAULT_STATE = new Map({
	notifications: new List(),
});

function NotificationContainer(props) {
	const state = (props.state || DEFAULT_STATE).toJS();
	return (
		<Notification
			leaveFn={i => props.deleteNotification(i)}
			notifications={state.notifications}
			autoLeaveError={props.autoLeaveError}
		/>
	);
}

NotificationContainer.displayName = 'Container(Notification)';
NotificationContainer.propTypes = {
	deleteNotification: PropTypes.func,
	autoLeaveError: PropTypes.bool,
	...cmfConnect.propTypes,
};

export default NotificationContainer;
