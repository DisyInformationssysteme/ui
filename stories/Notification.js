import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { Notification, IconsProvider } from '../src/index';

class NotificationWrapper extends React.Component {
	constructor() {
		super();
		this.notifications = [
		];
		this.counter = 1;
		setTimeout(() => {
			this.notifications = this.notifications.concat([
				{
					id: 'story-1',
					message: 'This is a feedback of your operation1, This is a feedback of your operation1',
					action: {
						label: 'Haha',
						icon: 'talend-undo',
						onClick: action('click Haha'),
					},
				},
			]);
			this.setState({	a: 'A' });
		}, 500);
		setTimeout(() => {
			this.notifications = this.notifications.concat([
				{
					id: 'story-2',
					type: 'error',
					message: 'This is a feedback of your operation2, This is a feedback of your operation1, This is a feedback of your operation1',
					action: {
						label: 'undo',
						icon: 'talend-undo',
						onClick: action('click undo'),
					},
				},
			]);
			this.setState({	b: 'B' });
		}, 1000);
		setTimeout(() => {
			this.notifications = this.notifications.concat([
				{
					id: 'story-3',
					type: 'warning',
					message: 'This is a feedback of your operation3',
				},
			]);
			this.setState({	b: 'B' });
		}, 2000);
		setTimeout(() => {
			this.notifications = this.notifications.concat([
				{
					id: 'story-4',
					type: 'warning',
					message: 'This is a feedback of your operation4',
				},
			]);
			this.setState({	b: 'B' });
		}, 2500);
	}
	leaveFn(notification) {
		const index = this.notifications.indexOf(notification);
		if (index !== -1) {
			this.notifications.splice(index, 1);
			this.setState({ count: this.counter + 1 });
		}
	}
	render() {
		return (
			<Notification
				leaveFn={i => this.leaveFn(i)}
				notifications={this.notifications}
			/>
		);
	}
}

storiesOf('Notification', module)
	.addWithInfo('default', () => (
		<nav>
			<h1>Notification</h1>
			<IconsProvider />
			<h2>Definition</h2>
			<p>The Notification component display </p>
			<h2>Examples</h2>
			<NotificationWrapper />
		</nav>
	));
