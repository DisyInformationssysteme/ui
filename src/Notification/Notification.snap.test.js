import React from 'react';
import renderer from 'react-test-renderer';

import Notification from './Notification.component';

describe('Notification', () => {
	it('should render', () => {
		const notifications = [
			{
				id: 0,
				message: 'This is a feedback of your operation0, This is a feedback of your operation0',
				action: {
					label: 'UNDO',
					icon: 'talend-undo',
					onClick: () => {},
				},
			},
			{
				id: 1,
				type: 'info',
				message: 'This is a feedback of your operation1, This is a feedback of your operation1',
				action: {
					label: 'UNDO',
					icon: 'talend-undo',
					onClick: () => {},
				},
			},
			{
				id: 2,
				type: 'error',
				message: 'This is a feedback of your operation2, This is a feedback of your operation2',
			},
			{
				id: 3,
				type: 'warning',
				message: 'This is a feedback of your operation3, This is a feedback of your operation3',
				action: {
					label: 'UNDO',
					icon: 'talend-undo',
					onClick: () => {},
				},
			},
		];
		const wrapper = renderer.create(
			<Notification notifications={notifications} leaveFn={() => {}} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
