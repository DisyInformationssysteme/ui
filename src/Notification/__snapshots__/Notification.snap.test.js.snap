exports[`Notification should render 1`] = `
<div
  className="tc-notification-container">
  <div
    className="react-addons-css-transition-group">
    <div
      className="tc-notification tc-notification-info"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="status"
      tabIndex={2}>
      <button
        className="tc-notification-action .tc-notification-close tc-notification-action .tc-notification-close-default"
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button">
        <svg
          aria-hidden="true"
          className="tc-svg-icon"
          title={null}>
          <use
            xlinkHref="#talend-cross-big" />
        </svg>
        <span>
          
        </span>
      </button>
      <p
        className="tc-notification-message">
        This is a feedback of your operation0, This is a feedback of your operation0
        <button
          className="tc-notification-action tc-notification-message-action tc-notification-action tc-notification-message-action-default"
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button">
          <svg
            aria-hidden="true"
            className="tc-svg-icon"
            title={null}>
            <use
              xlinkHref="#talend-undo" />
          </svg>
          <span>
            UNDO
          </span>
        </button>
      </p>
      <div
        className="tc-notification-timer-bar" />
    </div>
    <div
      className="tc-notification tc-notification-info"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="status"
      tabIndex={2}>
      <button
        className="tc-notification-action .tc-notification-close tc-notification-action .tc-notification-close-default"
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button">
        <svg
          aria-hidden="true"
          className="tc-svg-icon"
          title={null}>
          <use
            xlinkHref="#talend-cross-big" />
        </svg>
        <span>
          
        </span>
      </button>
      <p
        className="tc-notification-message">
        This is a feedback of your operation1, This is a feedback of your operation1
        <button
          className="tc-notification-action tc-notification-message-action tc-notification-action tc-notification-message-action-default"
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button">
          <svg
            aria-hidden="true"
            className="tc-svg-icon"
            title={null}>
            <use
              xlinkHref="#talend-undo" />
          </svg>
          <span>
            UNDO
          </span>
        </button>
      </p>
      <div
        className="tc-notification-timer-bar" />
    </div>
    <div
      className="undefined tc-notification tc-notification-error"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="status"
      tabIndex={2}>
      <button
        className="tc-notification-action .tc-notification-close tc-notification-action .tc-notification-close-default"
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button">
        <svg
          aria-hidden="true"
          className="tc-svg-icon"
          title={null}>
          <use
            xlinkHref="#talend-cross-big" />
        </svg>
        <span>
          
        </span>
      </button>
      <p
        className="tc-notification-message">
        This is a feedback of your operation2, This is a feedback of your operation2
      </p>
    </div>
    <div
      className="tc-notification tc-notification-warning"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="status"
      tabIndex={2}>
      <button
        className="tc-notification-action .tc-notification-close tc-notification-action .tc-notification-close-default"
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button">
        <svg
          aria-hidden="true"
          className="tc-svg-icon"
          title={null}>
          <use
            xlinkHref="#talend-cross-big" />
        </svg>
        <span>
          
        </span>
      </button>
      <p
        className="tc-notification-message">
        This is a feedback of your operation3, This is a feedback of your operation3
        <button
          className="tc-notification-action tc-notification-message-action tc-notification-action tc-notification-message-action-default"
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button">
          <svg
            aria-hidden="true"
            className="tc-svg-icon"
            title={null}>
            <use
              xlinkHref="#talend-undo" />
          </svg>
          <span>
            UNDO
          </span>
        </button>
      </p>
      <div
        className="tc-notification-timer-bar" />
    </div>
  </div>
</div>
`;
